res/test_generate_update_metadata.py

```
@@ -0,0 +1,162 @@
+        self.assertNotEqual(
+            self.verify(metadata, signature, [artifact], wrong_key).returncode,
+            0,
+        )
+
+    def test_sign_rejects_invalid_release_inputs(self):
+        artifact = self.artifact()
+        spec = [("windows", "x86_64", "exe", artifact)]
+        self.assertNotEqual(self.sign(spec, version="1.4.7")[2].returncode, 0)
+        self.assertNotEqual(self.sign(spec, release_id="bad/tag")[2].returncode, 0)
+        self.assertNotEqual(self.sign(spec, seed="invalid")[2].returncode, 0)
+        duplicate = spec + [("windows", "x86_64", "exe", artifact)]
```

Ruff (0.16.1)

[warning] Consider `[*spec, ("windows", "x86_64", "exe", artifact)]` instead of concatenation

Replace with `[*spec, ("windows", "x86_64", "exe", artifact)]`

(RUF005)